evm/src/cpu/kernel/asm/transactions/router.asm

```diff
@@ -34,3 +37,30 @@ read_txn_from_memory:
     // At this point, since it's not a type 1 or 2 transaction,
     // it must be a legacy (aka type 0) transaction.
     %jump(process_type_0_txn)
+
+global update_txn_trie:
+    // stack: txn_rlp_len, txn_counter, num_nibbles, retdest
+    // We need to copy the transaction rlp since otherwise
+    // it will be eventually overwritten when hashing the
+    // state trie. We will write in the trie data segment.
```

Comment:
I'm a bit confused by this comment. Isn't copying to `SEGMENT_TRIE_DATA` necessary in any case since `mpt_insert` expects data there?